sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -207,6 +207,14 @@ class SessionCatalog(
           "you cannot create a database with this name.")
     }
     validateName(dbName)
+    // SPARK-25464 fail if DB location exists and is not empty
+    val dbPath = new Path(dbDefinition.locationUri)
+    val fs = dbPath.getFileSystem(hadoopConf)
+    if (!externalCatalog.databaseExists(dbName) && fs.exists(dbPath)
+      && fs.listStatus(dbPath).nonEmpty) {
+      throw new AnalysisException(
+        s"Cannot create database at location $dbPath because the path is not empty.")
```

Comment:
This says 'path not empty' but that suggests 'path is not the empty string'. This means 'the path exists already' right?

Reply:
yes , it means path already exists, I will update the message accordingly

Reply:
If I change the message it will lead to confusion.
for e.g.
User can be able to create database to the location as long as the path does not exists or it should be empty.

If I say path already exits it will confuse, so I wanted to highlight the path is not empty

Reply:
Are you trying to describe a case where the path is a directory that exists, but contains files? or a path that does not exist at all? either way I think that type of description is better. An empty path implies something different from empty directory.

Reply:
yes I wanted to describe a case where the path is a directory that exists but contains files. if the above description is correct then I will update it